### ISSUE TITLE
Maybe provide a parameter to avoid browser caching for PyPI resolution requests

### ISSUE THREAD

**Issue by CNSeniorious000** [pyodide/micropip#122]
Currently micropip passes a kwargs to `pyfetch` during resolution.

https://github.com/pyodide/micropip/blob/5d4a8696f7daf302f060ffded34f9a84455fcffc/micropip/install.py#L138

I think it is reasonable to add a `no_cache` parameter to `install` command or the `Transaction` class to skip browser caching. Currently it is not possible to skip browser caching for PyPI simple API calls.

**Comment by ryanking13** [comment]
Thanks @CNSeniorious000! I can think of two ways to implement this.

1. adding a `cache` parameter as you suggest.
2. adding a `headers` parameter that users can pass arbitrary request headers when calling fetch. For instance, people can do `headers={'cache': 'no-store'}` or `headers={'cache': "no-cache"}` to pass cache-related parameters.

The second option would provide more flexibility, but it would also leak the internal implementation of `micropip.install`. WDYT?

**Comment by CNSeniorious000** [comment]
But `cache: "no-cache"` parameter is `fetch`'s kwargs instead of the headers. Do you mean expose a `**fetch_kwargs` in `micropip.install`?

And then here we don't need to init the `fetch_kwargs` with an empty dict:

https://github.com/pyodide/micropip/blob/5d4a8696f7daf302f060ffded34f9a84455fcffc/micropip/install.py#L116-L118

**Comment by ryanking13** [comment]
Oh yes, `fetch_kwargs` is what I meant, not `headers`.

**Comment by CNSeniorious000** [comment]
One more question. If I specify `cache='no-cache'` in **fetch_kwargs, both the resolution step and the downloading step will skip the cache. But I think `whl` urls should be fetched with `force-cache` instead. Is there a way to allow customizing the two fetch kwargs separately?

**Comment by ryanking13** [comment]
> Is there a way to allow customizing the two fetch kwargs separately?

I don't think we have such an option now, but I think it is reasonable. This may be something that we need to consider when re-designing the micropip module (https://github.com/pyodide/micropip/issues/112) cc: @RulerOfCakes
